pkg/reconciler/broker/config.go

```diff
@@ -18,46 +18,52 @@ package broker
 
 import (
 	"context"
+	"encoding/json"
 	"errors"
-	"strings"
+	"fmt"
+	"github.com/ghodss/yaml"
+	"go.uber.org/zap"
 
 	corev1 "k8s.io/api/core/v1"
-	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-	"k8s.io/apimachinery/pkg/runtime"
 	messagingv1beta1 "knative.dev/eventing/pkg/apis/messaging/v1beta1"
+	"knative.dev/eventing/pkg/logging"
 )
 
 type Config struct {
 	DefaultChannelTemplate messagingv1beta1.ChannelTemplateSpec
 }
 
+const (
+	channelTemplateSpec = "channelTemplateSpec"
+)
+
 func NewConfigFromConfigMapFunc(ctx context.Context) func(configMap *corev1.ConfigMap) (*Config, error) {
 	return func(configMap *corev1.ConfigMap) (*Config, error) {
+		config := &Config{
+			DefaultChannelTemplate: messagingv1beta1.ChannelTemplateSpec{},
+		}
 
-		apiVersion, ok := configMap.Data["channelTemplateSpec.apiVersion"]
-		if !ok {
-			return nil, errors.New("channelTemplateSpec.apiVersion not found in config")
+		temp, present := configMap.Data[channelTemplateSpec]
```

Comment:
Should the validation be done by the config validation webhook? We might want to have it in both places.

Reply:
I feel like the second validation is a waste, guard code access for nils and stuff but don't validate the values. because if you change the defaults you have to redeploy two times